Sources/SwiftParser/Availability.swift

```diff
@@ -253,6 +253,27 @@ extension Parser {
     return self.consumeAnyToken()
   }
 
+  /// Consume the unexpected version token(e.g. integerLiteral, floatingLiteral, identifier) until the period no longer appears.
+  private mutating func parseUnexpectedVersionTokens() -> RawUnexpectedNodesSyntax? {
+    var unexpectedTokens: [RawSyntax] = []
+    var keepGoing: RawTokenSyntax? = nil
+    var loopProgress = LoopProgressCondition()
+    repeat {
+      if let keepGoing {
+        unexpectedTokens.append(RawSyntax(keepGoing))
+      }
+      if let unexpectedVersion = self.consume(if: .integerLiteral, .floatingLiteral, .identifier) {
+        unexpectedTokens.append(RawSyntax(unexpectedVersion))
+      }
+      keepGoing = self.consume(if: .period)
+    } while keepGoing != nil && loopProgress.evaluate(currentToken)
+
+    guard !unexpectedTokens.isEmpty else {
+      return nil
+    }
+    return RawUnexpectedNodesSyntax(elements: unexpectedTokens, arena: self.arena)
```

Comment:
You can simplify this slightly to the following. You can also change `unexpectedTokens` to `[RawTokenSyntax]`.
```suggestion
    return RawUnexpectedNodesSyntax(unexpectedTokens, arena: self.arena)
```

This uses the following convenience initializer.

https://github.com/apple/swift-syntax/blob/1555299f9e1fac77cf92677dae4dc8b46e9aa3d0/Sources/SwiftParser/SyntaxUtils.swift#L39-L43

Reply:
I've changed both initializers, thank you for your kind feedback!